aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/DefaultStreamingStrategy.java

```diff
@@ -98,14 +92,8 @@ private boolean stream(Entity entity, Emitter emitter) {
         //Gather children and in the condition they are ready to stream, add them to the streamable list.
```

Comment:
I think/fear a different race condition just moved 2 lines up now by removing the subsegmentLock.  And you will hit #181  again.

As the copy of `entity.getSubsegments()` is taken outside the protected region.
So it is possible that an entity is added while the subsegment list is being copied what can/will result in a `ConcurrentModificationException` being thrown.

Possible solutions I see:
* Make the collection returned by `getSubsegments` thread safe 
* Let `getSubsegments` return an (immutable) copy of the list (or provide another function which does).

Reply:
Thanks for noticing this @steven-aerts - I've updated to return a copy